src/jvmLogbackTest/kotlin/io/github/oshai/kotlinlogging/logback/internal/LogbackLoggerWrapperTest.kt

```diff
@@ -0,0 +1,421 @@
+              when (given.level) {
+                "TRACE" -> logger.trace(marker, given.message)
+                "DEBUG" -> logger.debug(marker, given.message)
+                "INFO" -> logger.info(marker, given.message)
+                "WARN" -> logger.warn(marker, given.message)
+                "ERROR" -> logger.error(marker, given.message)
+              }
+            }
+            thenExpect { given ->
+              level = given.level
+              marker = given.marker
+              name = loggerName
+              message = given.message
+            }
+          }
+          test("$levelValue SLF4J API with marker and arg1") {
+            givenStatement {
+              level = levelValue.toString()
+              marker = "marker"
+              messagePrefix = "SLF4J message:"
+              message = "$messagePrefix {}"
+              arg1 = "val1"
+            }
+            whenCalling { given ->
+              val marker = KMarkerFactory.getMarker(given.marker)
+              when (given.level) {
+                "TRACE" -> logger.trace(marker, given.message, given.arg1)
+                "DEBUG" -> logger.debug(marker, given.message, given.arg1)
+                "INFO" -> logger.info(marker, given.message, given.arg1)
+                "WARN" -> logger.warn(marker, given.message, given.arg1)
+                "ERROR" -> logger.error(marker, given.message, given.arg1)
+              }
+            }
+            thenExpect { given ->
+              level = given.level
+              marker = given.marker
+              name = loggerName
+              message = "${given.messagePrefix} ${given.arg1}"
+            }
+          }
+          test("$levelValue SLF4J API with marker and arg1 arg2") {
+            givenStatement {
+              level = levelValue.toString()
+              marker = "marker"
+              messagePrefix = "SLF4J message:"
+              message = "$messagePrefix {} {}"
+              arg1 = "val1"
+              arg2 = "val2"
+            }
+            whenCalling { given ->
+              val marker = KMarkerFactory.getMarker(given.marker)
+              when (given.level) {
+                "TRACE" -> logger.trace(marker, given.message, given.arg1, given.arg2)
+                "DEBUG" -> logger.debug(marker, given.message, given.arg1, given.arg2)
+                "INFO" -> logger.info(marker, given.message, given.arg1, given.arg2)
+                "WARN" -> logger.warn(marker, given.message, given.arg1, given.arg2)
+                "ERROR" -> logger.error(marker, given.message, given.arg1, given.arg2)
+              }
+            }
+            thenExpect { given ->
+              level = given.level
+              marker = given.marker
+              name = loggerName
+              message = "${given.messagePrefix} ${given.arg1} ${given.arg2}"
+            }
+          }
+          test("$levelValue SLF4J API with marker and varargs") {
```

Comment:
do you think that those tests can be changed to regular test functions with annotations? I am not sure it will be easy to maintain those, and tools will have suboptimal experience with those. If it's for the multiple levels and given that it's unit tests, I am not sure we have to tests all levels for each scenario.

Reply:
Good point, I simplified that complexity, have a look.